### ISSUE TITLE
@psalm-assert creates MixedAssignment based on assertion order

### ISSUE THREAD

**Issue by Shyvno** [vimeo/psalm#9101]
Asserting the type of array followed up by asserting the contents of the array creates a MixedAssignment error
https://psalm.dev/r/aaa9efb258

Switching the order of assertions works:
https://psalm.dev/r/ca7ae838e6

**Comment by annervisser** [comment]
Could be the same issue as https://github.com/vimeo/psalm/issues/9078

**Comment by orklah** [comment]
Indeed. It seems the assertion is applied onto the main part of the array but not on extra keys when the array is not sealed.

@danog if you find time to look at this it would be great :)
